### PR TITLE
TerrainNode refactor

### PR DESCRIPTION
**Pull request by jim-martin**
Cleans up some internals of TerrainNode without effecting any public APIs.

1. Update some variable names and code styling to improve readability.
1. Factor out math related to zoom level calculations.
1. Eliminate some code duplication between ImageAPI and TerrainNode.
1. Store some common parameters in a 'constants' struct.
1. Address #12.
1. Load style and terrain texture in parallel.

### CODE REVIEW

**Review by avi-c — COMMENTED** [review]
Have you tested the examples and installation via Carthage and CocoaPods?

File: Examples/DemoPlacementViewController.swift [hunk 1]
```diff
@@ -30,8 +30,11 @@ class DemoPlacementViewController: UIViewController {
         sceneView.showsStatistics = true
 
         //Set up initial terrain and materials
-        let terrainNode = TerrainNode(minLat: 50.044660402821592, maxLat: 50.120873988090956,
-                                      minLon: -122.99017089272466, maxLon: -122.86824490727534)
+        let southWest = CLLocation(latitude: 50.044660402821592, longitude: -122.99017089272466)
+        let northEast = CLLocation(latitude: 50.120873988090956, longitude: -122.86824490727534)
+//        let terrainNode = TerrainNode(minLat: 50.044660402821592, maxLat: 50.120873988090956,
```
Comment (ghost) [inline Examples/DemoPlacementViewController.swift]: Delete this commented-out line

File: MapboxSceneKit/Math.swift [hunk 2]
```diff
@@ -40,12 +42,27 @@ internal class Math {
...
         let bottomRight = tile2LatLng(x: x + 1, y: y + 1, z: z)
+        
         return ((bottomRight.lat, topLeft.lat), (topLeft.lon, bottomRight.lon))
     }
+    
+    static func zoomLevelForBounds(southWestCorner: CLLocation, northEastCorner: CLLocation) -> Int {
+        let distance = northEastCorner.distance(from: southWestCorner) / 1000.0 //use kilometers
+        let imageSize = Double(Constants.maxTextureImageSize)
+        let latitudeAdjustment = cos(.pi * northEastCorner.coordinate.latitude / 180)
+        let arg = Constants.earthDiameterInKilometers
```
Comment (ghost) [inline MapboxSceneKit/Math.swift:60]: @avi-c: @jim-martin mentioned that you had gotten this formula from... somewhere. Do you happen to remember where? Mostly asking so we can think of a better name than "arg"

**Review by ghost — APPROVED** [review]
Just minor comments. This looks great 🚀

**Pull request merged**